pkg/generate/app/sourcelookup.go

```diff
@@ -306,8 +308,10 @@ func (r *SourceRepository) DetectAuth() error {
 			fmt.Sprintf("SystemRoot=%s", os.Getenv("SystemRoot")),
 		)
 	}
+
 	gitRepo := git.NewRepositoryWithEnv(env)
-	_, err = CloneAndCheckoutSources(gitRepo, url.StringNoFragment(), url.URL.Fragment, tempSrc, "")
+	glog.V(4).Infof("Checking if %v requires authentication", url.StringNoFragment())
+	_, _, err = gitRepo.TimedListRemote(10*time.Second, url.StringNoFragment(), "")
```

Comment:
so this will respect HTTP_PROXY/etc env variables, presumably.  does it also respect proxy configuration supplied in a .gitconfig, assuming that's a thing that can be done?

Reply:
@bparees it won't and I don't know that we want to go through the trouble of understanding your git environment. There's a few ways you can specify the proxy... you can use env vars, a global http.proxy setting, or a per-domain setting like: http.[my-domain].proxy. I discussed this with @coreydaley yesterday and it didn't seem worth the work just to decide whether we should display a warning about possibly requiring secrets for your build config.

Reply:
We agreed that a better experience is simply to have it fail fast and if it didn't work, then the warning can be safely ignored.

Reply:
@csrwng fair enough, especially since we're timing this command out after 10s now, so it won't hang for a long time.